### PR TITLE
Externalize JavaScript: hqadmin pages

### PR DESCRIPTION
**Pull request by orangejenny**
@millerdev

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 71 errors:

* corehq/apps/hqadmin/static/hqadmin/js/admin_restore.js, line 3 - 'myCodeMirror' is defined but never used (no-unused-vars)
'CodeMirror' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/admin_restore.js, line 12 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/couch_changes.js, line 6 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/couch_changes.js, line 7 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/couch_changes.js, line 8 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/couch_changes.js, line 9 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/couch_changes.js, line 14 - 'd3' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/couch_changes.js, line 17 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 65 - 'ws4redis' is defined but never used (no-unused-vars)
'WS4Redis' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 98 - 'd3' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 106 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 115 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 116 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 117 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 118 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 124 - 'd3' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 126 - 'nv' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 144 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 173 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 179 - Expected '!==' and instead saw '!='. (eqeqeq)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 199 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 209 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 236 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 242 - 'map' is defined but never used (no-unused-vars)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 278 - 'map' is defined but never used (no-unused-vars)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 295 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 316 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 322 - 'i' is already defined (no-redeclare)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 326 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 329 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/dimagisphere_helper.js, line 358 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/domain_faceted_report.js, line 6 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/domain_faceted_report.js, line 13 - Expected a conditional expression and instead saw an assignment. (no-cond-assign)
* corehq/apps/hqadmin/static/hqadmin/js/domain_faceted_report.js, line 28 - 'HQVisualizations' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/domain_faceted_report.js, line 38 - Expected indentation of 4 space characters but found 0. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/hqadmin_base_report.js, line 4 - 'HQReportDataTables' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/raw_couch.js, line 6 - 'myCodeMirror' is defined but never used (no-unused-vars)
'CodeMirror' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/raw_couch.js, line 15 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 13 - 'second_str' is already defined (no-redeclare)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 27 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 69 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 76 - 'error' is defined but never used (no-unused-vars)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 80 - Empty block statement. (no-empty)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 135 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 146 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 203 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 226 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 227 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 228 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 229 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 230 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 231 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 232 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 233 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 240 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 241 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 242 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 243 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 244 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 245 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 246 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 247 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 271 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 291 - Missing trailing comma. (comma-dangle)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 297 - 'alert_user' is not defined. (no-undef)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 301 - 'error' is defined but never used (no-unused-vars)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 305 - Empty block statement. (no-empty)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 313 - Missing semicolon. (semi)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 316 - 'DbComparisons' is defined but never used (no-unused-vars)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 332 - 'couchViewModel' is already defined (no-redeclare)
* corehq/apps/hqadmin/static/hqadmin/js/system_info.js, line 334 - 'couchViewModel' is already defined (no-redeclare)

**Comment by orangejenny** [comment]
ignoring dimagimon

File: corehq/apps/hqadmin/static/hqadmin/js/system_info.js [hunk 1]
```diff
@@ -0,0 +1,373 @@
+    function format_date(datestring) {
```
Comment (millerdev) [inline corehq/apps/hqadmin/static/hqadmin/js/system_info.js]: What's up with the extra level of indentation?

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqadmin/static/hqadmin/js/system_info.js (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/hqadmin/static/hqadmin/js/system_info.js]: oops, fixed

**Pull request merged**